upload-distribution.sh

```diff
@@ -16,21 +16,40 @@ fi
 
 echo "#####################################################"
 echo "# Uploading Hibernate $PROJECT $RELEASE_VERSION on"
-echo "# SourceForge: $DIST_PARENT_DIR"
+echo "# SourceForge: $REMOTE_DIST_PARENT_DIR"
 echo "#####################################################"
 echo "Workspace: $WORKSPACE"
 
 pushd $WORKSPACE
 
-((echo mkdir $DIST_PARENT_DIR/$RELEASE_VERSION; echo quit) | sftp -b - frs.sourceforge.net) || echo "Directory already exists. Skipping creation."
-scp README.md frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
-scp changelog.txt frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
-scp distribution/target/hibernate-$PROJECT-$RELEASE_VERSION-dist.zip frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
-scp distribution/target/hibernate-$PROJECT-$RELEASE_VERSION-dist.tar.gz frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
+((echo mkdir $REMOTE_DIST_PARENT_DIR/$RELEASE_VERSION; echo quit) | sftp -b - frs.sourceforge.net) || echo "Directory already exists. Skipping creation."
 
-MODULE=modules/target/hibernate-$PROJECT-modules-$RELEASE_VERSION-wildfly-10-dist.zip
-if [ -f $MODULE ]; then
-	scp $MODULE frs.sourceforge.net:$DIST_PARENT_DIR/$RELEASE_VERSION/
+REMOTE_DIST_URL=frs.sourceforge.net:$REMOTE_DIST_PARENT_DIR/$RELEASE_VERSION/
+
+scp README.md $REMOTE_DIST_URL
+scp changelog.txt $REMOTE_DIST_URL
+
+# Recursive upload of the dist directory (whose content is project-specific)
+DIST_DIR=distribution/target/dist
+if [ -d $DIST_DIR ]; then
+	# Cd to the dist directory to prevent scp from uploading a "dist" directory
+	pushd $DIST_DIR
+	scp -r . $REMOTE_DIST_URL
```

Comment:
Are we sure we don't have files we don't want to publish to SF.net?

Reply:
The `distribution/target/dist` directory doesn't exist right now, we will have to change our builds to create it and put whatever we want in it. So, yes.

Reply:
That's why I didn't just pick the `distribution/target` directory, by the way.